packages/core/src/sdk/geolocation/useGeolocation.ts

```diff
@@ -1,27 +1,43 @@
 import { useEffect } from 'react'
 
 import { deliveryPromise } from 'discovery.config'
-import { useSession, validateSession, sessionStore } from 'src/sdk/session'
+import { validateSession, sessionStore } from 'src/sdk/session'
 
-export default function useGeolocation() {
-  const { isValidating: _, ...session } = useSession()
+const TIME_TO_VALIDATE_SESSION = 3000
+
+async function askGeolocationConsent() {
+  const { postalCode: stalePostalCode, geoCoordinates: staleGeoCoordinates } =
+    sessionStore.read()
```

Comment:
question: why change from `useSession()` to `sessionStore.read()`?

Reply:
We can't have `useSession` values updated since the execution of `useGeolocation`'s effect is one-time only, that's I was using stale values just to the first check then reading session values again.